src/mpid/ch4/shm/posix/eager/fbox/fbox_init.h

```diff
@@ -138,7 +198,8 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_eager_finalize()
     MPL_free(MPIDI_POSIX_eager_fbox_control_global.mailboxes.out);
     MPL_free(MPIDI_POSIX_eager_fbox_control_global.first_poll_local_ranks);
 
-    mpi_errno = MPIDU_shm_seg_free(MPIDI_POSIX_eager_fbox_control_global.shm_ptr[0]);
+    for (int i = 0; i < MPIDI_POSIX_eager_fbox_control_global.num_seg; i++)
```

Comment:
I am not sure if `for (int i ...)` is okay in MPICH. Maybe it is better to declare `int i;` separately?

Reply:
Since it is part of C99 and mpich now requires C99, let's allow it. In fact, I would like to promote using loop-local iterators. It is so much more readable. Similarly, we should declare temporary local variables close to where it is being used.